src/utils.ts

```diff
@@ -427,17 +427,20 @@ export function processEscape(str: string) {
  * @param originalClassScope the original ClassScope to be specialized
  * @param parent the parent of the original ClassScope
  * @param classType the new class type corresponding to specialized ClassScope: TSClass => ClassScope
+ * @param newName the name of new ClassScope
  * @returns a new specialized ClassScope
  */
 export function createClassScopeByClassType(
     originalClassScope: ClassScope,
-    parent: Scope,
+    parent: Scope | null,
```

Comment:
In which situation, `parent` will be `null`?

Reply:
done.
This does seem redundant and wrong.